hooks/500-create-xdg.chroot

```diff
@@ -0,0 +1,30 @@
+#!/bin/sh -ex
+
+echo "I: Creating xdg helper"
+
+mv /usr/bin/xdg-open /usr/bin/xdg-open-original
+cat >/usr/bin/xdg-open <<'EOF'
+#!/bin/sh
+exec snapctl user-open "$@"
+EOF
+chmod 755 /usr/bin/xdg-open
+
+# corresponding .desktop entry, needed for mimetype registration
+mkdir -p /usr/share/applications
+cat >/usr/share/applications/xdg-open.desktop <<EOF
+[Desktop Entry]
+Version=1.0
+Name=Url Handler Script
+Exec=/usr/bin/xdg-open %u
+MimeType=x-scheme-handler/http;x-scheme-handler/https;x-scheme-handler/mailto;x-scheme-handler/help;
+Type=Application
+EOF
+
+# define xdg-open as the default handler for common types
+cat >/usr/share/applications/mimeapps.list <<EOF
+[Default Applications]
+x-scheme-handler/http=xdg-open.desktop
+x-scheme-handler/https=xdg-open.desktop
+x-scheme-handler/mailto=xdg-open.desktop
+x-scheme-handler/help=xdg-open.desktop
+EOF
```

Comment:
This looks like it can still cause an infinite loop. Consider the following scenario:

1. app within ubuntu-desktop-session snap tries to open a web link.
2. `mimeapps.list` says the default handler for https links is `xdg-open.desktop`
3. `xdg-open.desktop` calls `/usr/bin/xdg-open link`, which makes a D-Bus call to `snap userd`
4. Outside confinement, `snap userd` calls `xdg-open-original link`.
5. `xdg-open-original` calls `gio open link`
6. `gio open` checks `mimeapps.list`, which lists the default handler for https links as `xdg-open.desktop`
7. `gio open` calls `/usr/bin/xdg-open link`, which makes a D-Bus call to `snap userd`
8. Go to step 4.

It's one step removed from the loop I mentioned in #24, but it is still there.

If we pick a different application as a scheme handler (e.g. firefox.desktop), then we don't call the xdg-open proxy but break in a different way. We end up using the `desktop-launch` D-Bus API, which doesn't currently support launching with a URL/filename. So in that case we just end up starting the browser without the link.

Reply:
Ok, so that's why before this patch, clicking in a link opened the browser without showing the page...

I tried putting directly the original xdg-open script, but didn't work. That's why I used this convoluted form.